server/src/packages/pas-form/applicants/applicants.controller.ts

```diff
@@ -32,38 +58,107 @@ export class ApplicantsController {
+    const { target_entity } = body;
+    let allowedAttrs;
 
-    await this.crmService.update('dcp_applicantinformations', id, allowedAttrs);
+    if (target_entity === 'dcp_applicantinformation') {
+      // we can't send target_entity to CRM because it doesn't exist there
+      allowedAttrs = pick(body, APPLICANT_ATTRIBUTES.filter(attribute => attribute !== 'target_entity'));
+
+      await this.crmService.update(
+        'dcp_applicantinformations',
+        id,
+        allowedAttrs,
+      );
+    } else if (target_entity === 'dcp_applicantrepresentativeinformation') {
+      allowedAttrs = pick(body, APPLICANT_REPRESENTATIVE_ATTRIBUTES.filter(attribute => attribute !== 'target_entity'));
+
+      // strip the hacky prepended "representative-" to use the exact CRM id
+      const representativeId = id.replace('representative-', '');
+
+      await this.crmService.update(
+        'dcp_applicantrepresentativeinformations',
+        representativeId,
+        allowedAttrs,
+      )
+    }
 
+    // regardless of which entity, return same response back to requesting client
     return {
       dcp_applicantinformationid: id,
-      ...body,
-    }
+      allowedAttrs,
+    };
   }
 
   @Post('/')
   create(@Body() body) {
-    const allowedAttrs = pick(body, APPLICANT_ATTRIBUTES);
+    const { target_entity } = body;
 
-    if (body.pas_form) {
-      return this.crmService.create('dcp_applicantinformations', {
-        ...allowedAttrs,
+    let allowedAttrs;
 
-        // Dy365 syntax for associating a newly-created record
-        // with an existing record.
-        // see: https://docs.microsoft.com/en-us/powerapps/developer/common-data-service/webapi/create-entity-web-api#associate-entity-records-on-create
-        'dcp_dcp_applicantinformation_dcp_pasform@odata.bind': [`/dcp_pasforms(${body.pas_form})`],
-      });
-    } else {
-      return this.crmService.create('dcp_applicantinformations', allowedAttrs);
+    // determine the appropraite attributes based on the entity type
+    if (target_entity === 'dcp_applicantinformation') {
+      allowedAttrs = pick(body, APPLICANT_ATTRIBUTES.filter(attribute => attribute !== 'target_entity'));
+
+      if (body.pas_form) {
+        return this.crmService.create('dcp_applicantinformations', {
+          ...allowedAttrs,
+          // Dy365 syntax for associating a newly-created record
+          // with an existing record.
+          // see: https://docs.microsoft.com/en-us/powerapps/developer/common-data-service/webapi/create-entity-web-api#associate-entity-records-on-create
+          'dcp_dcp_applicantinformation_dcp_pasform@odata.bind': [
+            `/dcp_pasforms(${body.pas_form})`,
+          ],
+        });
+      } else {
+        return this.crmService.create(
+          `dcp_applicantinformations`,
+          allowedAttrs,
+        );
+      }
+    } else if (target_entity === 'dcp_applicantrepresentativeinformation') {
```

Comment:
I was writing this thinking about future code maintainers, and was thinking that being explicit about the two branches would reinforce / make more transparent what's happening (vs. implicitly using a default target_entity with an `if` that handles one of the scenarios. 

@allthesignals do you think it would be clearer if the `if` statement was only for applicant representative, and the applicant entity was treated as default / outside of any conditional?

Reply:
The alternative I had in mind was this:

```
if (value === 1) { }
if (value === 2) { }
```

Reply:
@allthesignals if I see two `if`s like that, reading I would think that there would be a scenario where both conditionals are true and get executed.

Vs. in this case, the two conditionals are mutually exclusive so communicating that with the `if... else if` makes it (I think) more declarative.